cmd/jaeger/internal/integration/badger_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"testing"
+
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/plugin/storage/integration"
+)
+
+type BadgerStorageIntegration struct {
+	E2EStorageIntegration
+	logger *zap.Logger
+}
+
+func (s *BadgerStorageIntegration) initialize(t *testing.T) {
+	s.e2eInitialize(t)
+
+	s.Refresh = func(_ *testing.T) {}
+	s.CleanUp = s.cleanUp
+
+	s.logger = zap.NewNop()
+
+	// TODO: remove this once badger supports returning spanKind from GetOperations
+	s.GetOperationsMissingSpanKind = true
+	s.SkipArchiveTest = true
```

Comment:
since these are static, they should be assigned when creating BadgerStorageIntegration struct, not in the initialize function